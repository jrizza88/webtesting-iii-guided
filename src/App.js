import React, { Component } from 'react';
import './App.css';
import Speaker from './speaker/Speaker';

class App extends Component {
  state = {
    message: 'Nothing to say'
  }
  render() {
    return (
      <div className="App">
        {/* <header className="App-header">
          <p>
            Edit <code>src/App.js</code> and save to reload... if you dare, because the snappening is upon us!
            You press 'u' to update your changes if that is what you want!
          </p>
        </header> */}
        <Speaker message={this.state.message} speak={this.speak}/>
      </div>
    );
  }

  speak = () => {
    this.setState({ message: 'you are not mocking me!'})
  }
}

export default App;
