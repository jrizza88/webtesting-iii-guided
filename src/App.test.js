import React from 'react';
import {render, fireEvent } from 'react-testing-library';
import renderer from 'react-test-renderer'; // 1: install this npm module as a dev dependency

import App from './App';

describe('<App />', () => {
  // 2. write this test
  it('matches snapshot', () => {
    const tree = renderer.create(<App />); // generates a DOM tree

    // snapshots are a JSON representation of the DOM tree
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('is mocking me', () => {
    // on surface mock looks the same as 
    // const mock = () => {}; 
    // this is a mock function

    // there is almost a "secret part of this function"
    // as the below works differently from the above mock function
    // think of it as a totalitarian state
    const mock = jest.fn()


    const result = mock();
    mock();
    expect(result).toBeUndefined();
    expect(mock).toHaveBeenCalled();

    // you can get even more detailed
    expect(mock).toHaveBeenCalledTimes(2);
  });

  it('it is mocking me even more', () => {
      const mock = jest.fn(() => 'hello')

      const result = mock('smile');

      expect(result).toBe('hello')
      expect(mock).toHaveBeenCalled();
      expect(mock).toHaveBeenCalledTimes(1);
      expect(mock).toHaveBeenCalledWith('smile');
  })

  it('update message when speak is clicked', () => {
    const {getByText, queryByText} = render(<App />)

    getByText(/Nothing to say/i)

    expect(queryByText(/you are not mocking me!/i))
    .toBeFalsy();
  })

  it('message is updated when speak is clicked', () => {
    // const mock = jest.fn(() => '')
    const {getByText} = render(<App />)

    const button = getByText(/speak/i)

    fireEvent.click(button)

   getByText(/you are not mocking me!/i);

  })

});
