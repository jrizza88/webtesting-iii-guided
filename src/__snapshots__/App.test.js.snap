// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches snapshot 1`] = `
<div
  className="App"
>
  <button
    onClick={[Function]}
  >
    Speak
  </button>
  <div>
    Nothing to say
  </div>
  <button>
    Save
  </button>
  <button>
    Delete
  </button>
</div>
`;
